packages/core/src/config/index.ts

```diff
@@ -233,6 +236,19 @@ export async function normalizeUserCompilationConfig(
     config.define.FARM_HMR_PATH = normalizedDevServerConfig.hmr.path;
   }
 
+  if (
+    config.output.targetEnv !== 'node' &&
+    isArray(config.runtime.plugins) &&
+    !config.runtime.plugins.includes(ImportMetaPluginPath)
+  ) {
+    config.runtime.plugins.push(ImportMetaPluginPath);
+    config.define.FARM_PORT = String(normalizedDevServerConfig.port);
+    config.define.FARM_HOST =
```

Comment:
can't use hostname directly in this place. The default hostname is localhost. If he configured host true, then it would be wrong.

can use window.location.hostname directly at runtime